modules/apps/frontend-js/frontend-js-loader-support/src/main/java/com/liferay/frontend/js/loader/support/internal/JSLoaderSupportImpl.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.js.loader.support.internal;
+
+import com.liferay.frontend.js.loader.support.JSLoaderSupport;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.servlet.taglib.aui.ScriptData;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.io.IOException;
+import java.io.Writer;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(service = JSLoaderSupport.class)
+public class JSLoaderSupportImpl implements JSLoaderSupport {
+
+	@Override
+	public String getScriptBody(
+		String javascriptModule, String javascriptVariable,
+		String javascriptCode) {
+
+		StringBundler javascriptSB = new StringBundler(10);
+
+		javascriptSB.append("(function() {");
+
+		javascriptSB.append("window[");
+		javascriptSB.append("Symbol.for('__LIFERAY_WEBPACK_GET_MODULE__')]('");
+		javascriptSB.append(javascriptModule);
+		javascriptSB.append("').then((");
+		javascriptSB.append(javascriptVariable);
+		javascriptSB.append(") => {");
+		javascriptSB.append(javascriptCode);
+		javascriptSB.append("});");
+
+		javascriptSB.append("})();");
+
+		return javascriptSB.toString();
+	}
+
+	@Override
+	public void writeScript(
+		Writer writer, String javascriptModule, String javascriptVariable,
+		String javascriptCode) {
+
+		_writeScriptData(
+			writer,
+			getScriptBody(
+				javascriptModule, javascriptVariable, javascriptCode));
+	}
+
+	@Override
+	public void writeScript(
+		Writer writer, String javascriptModule, String javascriptVariable,
+		String javascriptCode, String template) {
```

Comment:
The template is needed to be able to surround the call to `getModule` with more code (see [this](https://github.com/liferay-frontend/liferay-portal/pull/696/files#diff-6288321e7c173fe0e442d4ed91d3fbc6c5b4a62c4e54b6a4b73d56e2781e4c5bR113)). I got the name from a similar feature we have in the bundler, which we use to wrap CSS AMD modules inside the code that writes the <link> node to the DOM.